src/compiler/parser.ts

```diff
@@ -436,6 +436,13 @@ module ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode: boolean = false;
 
+        export const enum StatementFlags {
+            None = 0,
+            Statement = 1,
+            ModuleElement = 2,
+            StatementOrModuleElement = Statement | ModuleElement
+        }

```

Comment:
We can't eliminate StatementOrModuleElement. We classify elements as either Statement or ModuleElement. We then want to sometimes only allow those classified as Statement and sometimes allow both classifications. If ModuleElements were also classified as Statements then there'd be no way to exclude them.